ql/src/java/org/apache/hadoop/hive/ql/exec/DataSketchesFunctions.java

```diff
@@ -235,12 +232,21 @@ public String getFunctionName() {
         return Optional.empty();
       } else {
         JavaTypeFactoryImpl typeFactory = new JavaTypeFactoryImpl(new HiveTypeSystemImpl());
+        Type type = returnType;
+        if (type instanceof ParameterizedType) {
+          ParameterizedType parameterizedType = (ParameterizedType) type;
+          if (parameterizedType.getRawType() == List.class) {
+          final RelDataType componentRelType = typeFactory.createType(parameterizedType.getActualTypeArguments()[0]);
+          return Optional
+              .of(typeFactory.createArrayType(typeFactory.createTypeWithNullability(componentRelType, true), -1));
+          }
+        }
         return Optional.of(typeFactory.createType(returnType));
       }
```

Comment:
yes; I also wanted to use this only temporarily - because this approach would not work for GenericUDF - but instead of hardcoding stuff; I think the best would be to create a `SqlReturnTypeInference`  which could translate the opbindings to things which could be processed by `GenericUDF#initialize`
opened: HIVE-23579